src/sage/combinat/cluster_algebra_quiver/mutation_type.py

```diff
@@ -342,12 +343,25 @@ def _connected_mutation_type(dg):
     n = dg.order()
     edges = dg.edges(sort=True)
     vertices = list(dg)
-    # initializing lists of the edges with labels (2, -1) or (1, -2); (4, -1) or (1, -4); or (2, -2), respectively
+    # initializing lists of the edges with labels (2, -1) or (1, -2);
+    # (4, -1) or (1, -4); or (2, -2), respectively
     exc_labels = []
     exc_labels41 = []
     double_edges = []
     # letter = None
 
+    # some short-circuits for rank 2
+    if n == 2 and len(edges) == 1:
+        ab = tuple(sorted(dg.edge_labels()[0]))
+        if ab == (-1, 1):
+            return QuiverMutationType(['A', 2])
+        if ab in {(-2, 1), (-1, 2)}:
+            return QuiverMutationType(['B', 2])
+        if ab in {(-3, 1), (-1, 3)}:
+            return QuiverMutationType(['G', 2])
+        if ab == (-2, 2):
+            return QuiverMutationType(['A', 1, 1])
```

Comment:
```suggestion
        return QuiverMutationType('R2', (-ab[0], ab[1]))
```
`QuiverMutationType` already handles all of these cases, plus it should handle the general case too.

Reply:
Thanks a lot. I have implemented your better fix.